Controller/CoreController.php

```diff
@@ -158,10 +155,25 @@ protected function getSearchHandler()
      */
     protected function getBaseTemplate()
     {
-        if ($this->getRequest()->isXmlHttpRequest()) {
+        if ($this->getCurrentRequest()->isXmlHttpRequest()) {
             return $this->getAdminPool()->getTemplate('ajax');
         }
 
         return $this->getAdminPool()->getTemplate('layout');
     }
+
+    /**
+     * Get the request object from the container.
+     *
+     * @return Request
+     */
+    private function getCurrentRequest()
+    {
+        // NEXT_MAJOR: simplify this when dropping sf < 2.4
```

Comment:
Guys, I no longer understand.

@core23 : I can't create a method named `getRequest`, because it already exists (and contains a deprecation). I can't stop calling `parent::getRequest` because I never call it.

`getCurrentRequest` is private on purpose. I want to have a convenience method for use in other methods of the class. It's not intended for usage by child classes, which should use the `$request` parameter.

@jordisala1991 we don't want people to get the request from the parent class anymore.